### PR TITLE
common : add image topic adapters

### PR DESCRIPTION
**Pull request by huiyujiang**
common : add image topic adapters

### CODE REVIEW

**Comment by ycool** [comment]
It fails the lint check. https://travis-ci.org/ApolloAuto/apollo/builds/309335100?utm_source=github_status&utm_medium=notification

File: modules/common/adapters/adapter.h [hunk 1]
```diff
@@ -364,7 +379,14 @@ class Adapter : public AdapterBase {
                 IdentifierType<::sensor_msgs::CompressedImage>) {
     return false;
   }
-
+  bool FeedFile(const std::string &message_file,
```
Comment (siyangy) [inline modules/common/adapters/adapter.h]: is this function needed?

File: modules/common/adapters/message_adapters.h [hunk 2]
```diff
@@ -62,6 +63,8 @@ using PadAdapter = Adapter<control::PadMessage>;
 using PerceptionObstaclesAdapter = Adapter<perception::PerceptionObstacles>;
 using PlanningAdapter = Adapter<planning::ADCTrajectory>;
 using PointCloudAdapter = Adapter<::sensor_msgs::PointCloud2>;
+using ImageShortAdapter = Adapter<::sensor_msgs::Image>;
+using ImageLongAdapter = Adapter<::sensor_msgs::Image>;
```
Comment (xiaoxq) [inline modules/common/adapters/message_adapters.h]: Simply define one ImageAdapter and use it for both.

File: modules/common/adapters/adapter_manager.h [hunk 3]
```diff
@@ -225,32 +225,35 @@ class AdapterManager {
...
-  REGISTER_ADAPTER(RelativeOdometry);
-  REGISTER_ADAPTER(InsStat);
-  REGISTER_ADAPTER(InsStatus);
-  REGISTER_ADAPTER(GnssStatus);
-  REGISTER_ADAPTER(SystemStatus);
-  REGISTER_ADAPTER(Mobileye);
-  REGISTER_ADAPTER(DelphiESR);
-  REGISTER_ADAPTER(ContiRadar);
-  REGISTER_ADAPTER(CompressedImage);
+ REGISTER_ADAPTER(Chassis);
```
Comment (xiaoxq) [inline modules/common/adapters/adapter_manager.h]: Please keep indents.

File: modules/common/adapters/adapter_manager.h [hunk 4]
```diff
@@ -208,7 +208,7 @@ class AdapterManager {
                                                    oneshot, autostart);
     } else {
       AWARN << "ROS timer is only available in ROS mode, check your adapter "
-               "config file! Return a dummy timer that won't function.";
+          "config file! Return a dummy timer that won't function.";
```
Comment (xiaoxq) [inline modules/common/adapters/adapter_manager.h]: Please keep indents.

File: modules/common/adapters/adapter.h [hunk 5]
```diff
@@ -239,7 +240,21 @@ class Adapter : public AdapterBase {
...
   }
-
+  /**
+   * @brief returns the most recent message pointer in the observing queue.
+   *
+   * /note
+   * Please call Empty() to make sure that there is data in the
+   * queue before calling GetLatestObservedPtr().
+   */
+  std::shared_ptr<D> GetLatestObservedPtr() const {
```
Comment (xiaoxq) [inline modules/common/adapters/adapter.h]: We'd better stick to using the const reference. Keep the "observed data" read-only.

File: modules/common/adapters/message_adapters.h (hunk 2 above)
Comment (ghdawn) [reply modules/common/adapters/message_adapters.h]: adapter is designed as a topic receiver instead of a sensor receiver, as is said by @siyangy

Comment (xiaoxq) [reply modules/common/adapters/message_adapters.h]: Oops, you are right :)

**Pull request merged**